Common/Securities/Positions/PositionManager.cs

```diff
@@ -0,0 +1,202 @@
+        /// keeping track of used buying power within a single time step to handle multiple calls to SetHoldings
+        /// in the same time step. At the end of the time step
+        /// </summary>
+        /// <param name="fill">The fill event -- this should only be events that change security holdings</param>
+        public void ProcessFill(OrderEvent fill)
+        {
+            if (fill.Status.IsFill())
+            {
+                ResolvePositionGroups();
+            }
+        }
+
+        /// <summary>
+        /// Resolves the algorithm's position groups from all of its holdings
+        /// </summary>
+        public void ResolvePositionGroups()
+        {
+            if (_requiresGroupResolution)
+            {
+                _requiresGroupResolution = false;
+                // TODO : Replace w/ special IPosition impl to always equal security.Quantity and we'll
+                // use them explicitly for resolution collection so we don't do this each time
+
+                Groups = ResolvePositionGroups(new PositionCollection(_securities
+                    .Where(kvp => kvp.Value.Invested).ToImmutableDictionary(
+                        kvp => kvp.Key,
+                        kvp => (IPosition) new Position(kvp.Value)
+                    )
+                ));
+            }
+        }
+
+        /// <summary>
+        /// Resolves position groups using the specified collection of positions
+        /// </summary>
+        /// <param name="positions">The positions to be grouped</param>
+        /// <returns>A collection of position groups containing all of the provided positions</returns>
+        public PositionGroupCollection ResolvePositionGroups(PositionCollection positions)
+        {
+            return _resolver.Resolve(positions);
+        }
+
+        /// <summary>
+        /// Determines which position groups could be impacted by changes in the specified positions
+        /// </summary>
+        /// <param name="positions">The positions to be changed</param>
+        /// <returns>All position groups that need to be re-evaluated due to changes in the positions</returns>
+        public IEnumerable<IPositionGroup> GetImpactedGroups(IReadOnlyCollection<IPosition> positions)
+        {
+            return _resolver.GetImpactedGroups(Groups, positions);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="PositionGroupKey"/> for the security's default position group
+        /// </summary>
+        public PositionGroupKey CreateDefaultKey(Security security)
+        {
+            return new PositionGroupKey(_defaultModel, security);
+        }
+
+        public IPositionGroup CreateDefaultGroup(Security security)
+        {
+            var key = CreateDefaultKey(security);
+            return Groups[key];
+        }
+
+        private void HoldingsOnQuantityChanged(object sender, SecurityHoldingQuantityChangedEventArgs e)
+        {
+            _requiresGroupResolution = true;
```

Comment:
Here we differ the calculation of the group resolution which seems different than what is being done on a `Fill`, wonder why? Should both be deferred instead?